sdk/helper/storagepacker/ids2keys.go

```diff
@@ -0,0 +1,99 @@
+package storagepacker
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/helper/cryptoutil"
+	"math"
+	"sort"
+	"strings"
+)
+
+type itemRequest struct {
+	// ID is the Item ID, provided by client
+	ID string
+
+	// Key is the identifier used internally, a hash of ID
+	Key string
+
+	// Value is the stored object, nil if not found
+	Value *Item
+}
+
+func GetItemIDHash(itemID string) string {
+	return hex.EncodeToString(cryptoutil.Blake2b256Hash(itemID))
+}
+
+// Length of key
+const KeyLength = 64
```

Comment:
Seems that we don't have to export this. Probably worth mentioning that this is the length returned by blake2b's hashing algo.

Reply:
You're correct that it's not strictly necessary. Let me think about it, it's not really the case that we need to export GetItemIdHash either. But, we should probably export both or neither.

The key is a little bit of a leaky abstraction; we still expose buckets (via GetBucket) and if you want to interact with a bucket you need to be able to create a key.